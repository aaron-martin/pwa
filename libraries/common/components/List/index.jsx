/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ListItem from './components/Item';
import styles from './style';

/**
 * The list component.
 * @param {Object} props - The component props.
 * @param {React.Children} props.children - Some content to display inside.
 */
class List extends Component {
  static Item = ListItem;

  static propTypes = {
    className: PropTypes.string,
  };

  static defaultProps = {
    className: '',
  };

  /**
   * Composes the props.
   * @returns {Object} The composed props.
   */
  getProps() {
    return {
      ...this.props,
      className: `${styles} ${this.props.className}`,
    };
  }

  /**
   * Renders the component.
   * @returns {JSX}
   */
  render() {
    return React.createElement('ul', this.getProps());
  }
}

export default List;
