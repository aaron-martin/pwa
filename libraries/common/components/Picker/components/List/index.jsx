/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './style';

/**
 * The default button for the Picker component.
 * @returns {JSX} The button component.
 */
const PickerList = ({ items, onClose, onSelect, selectedIndex }) => (
  <ul>
    {items.map((item, currentIndex) =>
      <li
        key={item.value}
        className={classNames({ [styles.active]: currentIndex === selectedIndex })}
      >
        <button
          className={styles.button}
          disabled={item.disabled}
          onClick={() => {
            onSelect(item.value);
            onClose();
          }}
        >
          {item.label}
        </button>
      </li>
    )}
  </ul>
);

PickerList.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  onSelect: PropTypes.func.isRequired,
  onClose: PropTypes.func,
  selectedIndex: PropTypes.number,
};

PickerList.defaultProps = {
  onClose: () => {},
  selectedIndex: null,
};

export default PickerList;
