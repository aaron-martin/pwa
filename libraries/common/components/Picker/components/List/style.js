/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { css } from 'glamor';

const button = css({
  display: 'block',
  width: '100%',
  padding: '10px 20px',
  outline: 'none',
  textAlign: 'left',
}).toString();

const active = css({
  button: {
    fontWeight: 'bold',
  },
}).toString();

export default {
  button,
  active,
};
