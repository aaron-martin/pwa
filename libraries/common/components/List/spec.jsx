/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { shallow } from 'enzyme';
import List from './index';

describe('<List />', () => {
  const children = [
    <List.Item key="0" />,
    <List.Item key="1" />,
    <List.Item key="2" />,
  ];

  it('renders with children', () => {
    const numChildren = children.length;
    const wrapper = shallow(<List>{children}</List>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(List.Item).length).toBe(numChildren);
  });

  it('renders without children', () => {
    const wrapper = shallow(<List />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(List.Item).length).toBe(0);
  });
});
