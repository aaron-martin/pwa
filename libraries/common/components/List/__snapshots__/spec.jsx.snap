// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders with children 1`] = `
<ul
  className="css-gkw1j5 "
>
  <ListItem
    className={null}
    isSelected={false}
    key="0"
  />
  <ListItem
    className={null}
    isSelected={false}
    key="1"
  />
  <ListItem
    className={null}
    isSelected={false}
    key="2"
  />
</ul>
`;

exports[`<List /> renders without children 1`] = `
<ul
  className="css-gkw1j5 "
/>
`;
